static/js/components/widgets/SiteContentEmbedField.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useCallback, SyntheticEvent } from "react"
+import { useSelector } from "react-redux"
+import { useRequest } from "redux-query-react"
+
+import { useWebsite } from "../../context/Website"
+import {
+  websiteContentListingRequest,
+  WebsiteContentListingResponse
+} from "../../query-configs/websites"
+import { getWebsiteContentListingCursor } from "../../selectors/websites"
+import { ContentListingParams } from "../../types/websites"
+import { useDebouncedState } from "../../hooks/state"
+
+interface Props {
+  insertEmbed: (id: string) => void
+  attach: string
+}
+
+export default function SiteContentEmbedField(props: Props): JSX.Element {
+  const { insertEmbed, attach } = props
+
+  // filterInput is to store user input and is updated synchronously
+  // so that the UI stays responsive
+  const [filterInput, setFilterInput] = useState("")
+  // filter, by contrast, is set by the setFilterDebounced function
+  // to cut down on extraneous requests. filter is set as a param in
+  // the listingParams object below, so updating the value causes the
+  // websiteContentListingRequest to be re-run, so we debounce for
+  // less silliness!
+  const [filter, setFilterDebounced] = useDebouncedState("", 300)
+
+  const onChangeFilter = useCallback(
+    (event: SyntheticEvent<HTMLInputElement>) => {
+      const newFilter = event.currentTarget.value
+      setFilterInput(newFilter)
+      setFilterDebounced(newFilter)
+    },
+    [setFilterDebounced]
+  )
+
+  const website = useWebsite()
+  const listingParams: ContentListingParams = {
+    name:   website.name,
+    type:   attach,
+    search: filter,
+    offset: 0
+  }
+
+  useRequest(websiteContentListingRequest(listingParams, false, false))
+
+  const listing: WebsiteContentListingResponse = useSelector(
+    getWebsiteContentListingCursor
+  )(listingParams)
+
+  return (
+    <div className="my-2 content-embed-widget">
+      <label>Resources</label>
+      <input
+        placeholder="filter"
+        type="text"
+        onChange={onChangeFilter}
+        value={filterInput}
+        className="form-control filter-input"
+      />
+      <div className="content-list">
+        {listing.results ?
+          listing.results.map(item => (
```

Comment:
I'm not sure if this is outside of the scope of this PR, but this list does not currently paginate.  I tested this both creating my own site and with an imported course.  The filtering function makes it usable because you can type what you're looking for, but it still only lists up to 10 results.  An indication of how many results were returned by your filter might be useful as well, but I think that's definitely out of scope.

Reply:
for the first pass I want to stick to the filtering alone, if we feel we want to add pagination later I think we can